test/e2e/upgrade_test.go

```diff
@@ -0,0 +1,90 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/test/txsim"
+	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/stretchr/testify/require"
+)
+
+// This will only run tests within the v1 major release cycle
+const MajorVersion = 1
+
+func TestMinorVersionCompatibility(t *testing.T) {
+	if os.Getenv("E2E") == "" {
```

Comment:
[question] I think this boolean will evaluate to true even if `E2E=false`. Potentially: 

```suggestion
	if os.Getenv("E2E") != "true" {
```

Reply:
Yeah, I could even combine `E2E` and `E2E_VERSIONS`

Reply:
SGTM